ecl/hql/hqlcollect.cpp

```diff
@@ -226,18 +238,26 @@ EclSourceType getEclSourceType(const char * tailname)
 static IIdAtom * deriveEclName(const char * filename)
 {
     if (!filename)
-        return NULL;
+        return nullptr;
+
+    StringBuffer tailname(pathTail(filename));
+    const char * ext = strrchr(tailname.str(), '.');
+#ifdef _USE_ZLIB
```

Comment:
Minor: Once code occurs 3 times, I wonder whether it should be commoned up.  E.g., in this case you could have

void removeZipExtension(StringBuffer & target, const char * source)

which would simplify the code - and would allow you to remove other extensions at a later date.